src/huggingface_hub/hub_mixin.py

```diff
@@ -406,44 +403,48 @@ def push_to_hub(
 
 
 class PyTorchModelHubMixin(ModelHubMixin):
-    def __init__(self, *args, **kwargs):
-        """
-        Mix this class with your torch-model class for ease process of saving &
-        loading from huggingface-hub.
-
-        Example:
-
-        ```python
-        >>> from huggingface_hub import PyTorchModelHubMixin
-
-
-        >>> class MyModel(nn.Module, PyTorchModelHubMixin):
-        ...     def __init__(self, **kwargs):
-        ...         super().__init__()
-        ...         self.config = kwargs.pop("config", None)
-        ...         self.layer = ...
-
-        ...     def forward(self, *args):
-        ...         return ...
-
-
-        >>> model = MyModel()
-        >>> model.save_pretrained(
-        ...     "mymodel", push_to_hub=False
-        >>> )  # Saving model weights in the directory
-        >>> model.push_to_hub(
-        ...     repo_id="mymodel", commit_message="model-1"
-        >>> )  # Pushing model-weights to hf-hub
+    """
+    Implementation of [`ModelHubMixin`] to provide model Hub upload/download
+    capabilities to PyTorch models. The model is set in evaluation mode by
+    default using `model.eval()` (dropout modules are deactivated). To train
+    the model, you should first set it back in training mode with
+    `model.train()`.
+
+    Example:
+
+    ```python
+    >>> import torch
```

Comment:
I changed the example to be something that users can copy-paste and actually run (the previous version was pseudo-code)